src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,109 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final Object[] EMPTY_ELEMENTDATA = {};
+    private Object [] elementData;
+    private int size;
+
+    public ArrayList(int initialCapacity) {
+        if (initialCapacity > 0) {
+            this.elementData = new Object[initialCapacity];
+        } else if (initialCapacity == 0) {
+            this.elementData = EMPTY_ELEMENTDATA;
+        } else {
+            throw new IllegalArgumentException("Illegal Capacity"
+                    + initialCapacity);
+        }
+    }
+
+    public ArrayList() {
+        this.elementData = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
+        int potentialSize = size + 1;
+        if (potentialSize > elementData.length) {
+            elementData = grow(elementData.length);
+        }
```

Comment:
```suggestion
        if (size == elementData.length) {
            grow(elementData.length);
        }
```

lets make `grow` method as void. And lets add `if (size == elementData.length)` check into `grow` method and rename it to `ensureCapacity`